character-controller.js

```diff
@@ -867,11 +867,12 @@ class StatePlayer extends PlayerBase {
     });
   }
   destroy() {
-    this.unbindState();
     this.appManager.unbindState();
 
     this.appManager.destroy();
-  
+    
+    this.unbindState();

```

Comment:
Can you make a detailed comment about why this needs to be done in this order? I think in general the unbind should happen first so that none of the destroy methods are changing app state during destruction.